### PR TITLE
feat(derive): `Past` batch validity variant

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a new variant to the `BatchValidity` enum that activates with Holocene. When a batch is outdated, after Holocene, the `Past` variant is returned. In the `BatchQueue`, this instructs a silent drop, _without_ flushing the active span batch and channel.

We should follow this PR up with removing the additional check within `add_batch`, deferring to the validity checks in `derive_next_batch`.

### CODE REVIEW

File: crates/derive/src/batch/validity.rs [hunk 1]
```diff
@@ -15,6 +15,9 @@ pub enum BatchValidity {
     Undecided,
     /// The batch may be valid, but cannot be processed yet and should be checked again later
     Future,
+    /// Introduced in Holocene, a special variant of the Drop variant that signals not to flush
+    /// the active batch and channel.
+    BatchOutdated,
```
Comment (sebastianst) [inline crates/derive/src/batch/validity.rs:20]: Following the others enum names, wouldn't you just name this `Outdated`? Note that I name the validity `past` in the specs. Or do you strongly prefer `outdated` over `past`? I thought that `past` pairs better with `future`.

File: crates/derive/src/stages/batch_queue.rs [hunk 2]
```diff
@@ -261,6 +270,9 @@ where
         if drop {
             self.prev.flush();
```
Comment (sebastianst) [inline crates/derive/src/stages/batch_queue.rs:271]: So do you now also flush pre-holocene?

File: crates/derive/src/batch/validity.rs (hunk 1 above)
Comment (clabby) [reply crates/derive/src/batch/validity.rs:20]: Not opinionated. We can standardized on the `Past` name.

File: crates/derive/src/stages/batch_queue.rs (hunk 2 above)
Comment (clabby) [reply crates/derive/src/stages/batch_queue.rs:271]: Good catch - no, we shouldn't be.

File: crates/derive/src/batch/validity.rs (hunk 1 above)
Comment (sebastianst) [reply crates/derive/src/batch/validity.rs:20]: Ok, also using `past` in the Go implementation now.

File: crates/derive/src/stages/batch_queue.rs (hunk 2 above)
Comment (sebastianst) [reply crates/derive/src/stages/batch_queue.rs:271]: does this also prevent flushing in the batchqueue itself?

**Comment by refcell** [comment]
This LGTM - rip coverage

File: crates/derive/src/stages/batch_queue.rs (hunk 2 above)
Comment (refcell) [reply crates/derive/src/stages/batch_queue.rs:271]: Yes it does - is that incorrect?

Comment (sebastianst) [reply crates/derive/src/stages/batch_queue.rs:271]: It also needs to be flushed if a batch is found to be invalid (`drop`).

**Pull request merged**